back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -13,7 +13,7 @@ class WebApi::V1::IdeasController < ApplicationController
   def json_forms_schema
     input = Idea.find params[:id]
     enabled_fields = IdeaCustomFieldsService.new(input.custom_form).enabled_fields
-    json_attributes = JsonFormsService.new.input_ui_and_json_multiloc_schemas enabled_fields, current_user, input.input_term
+    json_attributes = JsonFormsService.new.input_ui_and_json_multiloc_schemas enabled_fields, current_user, input.participation_method_on_creation, input.input_term
```

Comment:
The service would extract the participation method from the participation context of the custom form. This is exactly what `participation_method_on_creation` does.

Reply:
> from the participation context of the custom form

@sebastienhoorens Does this mean that the form object itself stores some information about the type of participation method? Or is this extracting the participation method based on the current phase? :thinking:

Reply:
@amanda-anderson The custom form is linked to either a project, or a phase (which is still called the participation context, but actually you now always need a phase to participate). This is then passed to `participation_method_for` in `factory.rb`.  The participation methods are separate classes, that are not persisted in the database.

For custom fields, there are two situations:
- Ideation: Then the participation context of the custom form is always the project, and when given a project `participation_method_for` will always create an ideation method
- Native survey: Then the participation context is always a phase and `participation_method_for` will create a native survey participation method.

Actually I'm not sure what would happen if an admin created an input in a voting phase (I think this is possible?). I'll take a look at this case to make sure this situation also still works.

Reply:
Yeah my only thought here was about admins adding inputs (to a specific phase) during other phases in the project :)

Reply:
Just tested it, and it still works the same for voting phases :)